src/main/java/org/fenixedu/academic/thesis/ui/service/ThesisProposalsService.java

```diff
@@ -161,9 +161,21 @@
     }
 
     public List<ThesisProposalsConfiguration> getThesisProposalsConfigurationsForCoordinator(User coordinator) {
-        return Degree.readBolonhaDegrees().stream().flatMap(degree -> degree.getExecutionDegrees().stream())
+        List<ThesisProposalsConfiguration> currentExecutionDegrees = Degree.readBolonhaDegrees().stream()
+                .flatMap(degree -> degree.getExecutionDegrees().stream())
                 .filter(executionDegree -> CoordinatorGroup.get(executionDegree.getDegree()).isMember(coordinator))
                 .flatMap(executionDegree -> executionDegree.getThesisProposalsConfigurationSet().stream()).distinct()
+                .collect(Collectors.toList());
+        
+        List<ThesisProposalsConfiguration> coordinatorExecutionDegrees =
```

Comment:
Instead of adding up to, this condition should replace the previous. Previously was getting the current coordinators of a degree, we want the coordinators for specific executions of a degree.